File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaPublicacionesBO/src/pe/edu/pucp/softbiblioteca/publicaciones/bo/TemaBO.java

/* [/]
 >> Project:    SoftBiblioecaPublicacionesBO
 >> File:       TemaBO.java
 >> Author:     Seguidores de VK
[/] */

package pe.edu.pucp.softbiblioteca.publicaciones.bo;
import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softbiblioteca.localidad.model.Biblioteca;
import pe.edu.pucp.softbiblioteca.publicaciones.dao.TemaDAO;
import pe.edu.pucp.softbiblioteca.publicaciones.dao.Tesis_TemaDAO;
import pe.edu.pucp.softbiblioteca.publicaciones.daoImpl.TemaDAOImpl;
import pe.edu.pucp.softbiblioteca.publicaciones.daoImpl.Tesis_TemaDAOImpl;
import pe.edu.pucp.softbiblioteca.publicaciones.model.Tema;
import pe.edu.pucp.softbiblioteca.publicaciones.model.Tesis;
import pe.edu.pucp.softbiblioteca.publicaciones.model.TipoGrado;
import pe.edu.pucp.softbiblioteca.util.VKF_Formatter;

public class TemaBO {
    private final TemaDAO temaDAO;
    private final VKF_Formatter vkf;
    private final Tesis_TemaDAO tesis_temaDAO;
    
    public TemaBO(){
        this.temaDAO = new TemaDAOImpl();
        this.tesis_temaDAO = new Tesis_TemaDAOImpl();
        this.vkf = new VKF_Formatter();
    }
    
    public Integer insertar(String titulo) {        
        Tema tema = new Tema(null, titulo);
        return this.temaDAO.insertar(tema);
    }
    
    public Integer modificar(Integer idTema, String titulo) {        
        Tema tema = new Tema(idTema, titulo);
        return this.temaDAO.modificar(tema);
    }
    
    public Integer eliminar(Integer idTema) {        
        Tema tema = new Tema();
        tema.setIdTema(idTema);
        return this.temaDAO.eliminar(tema);
    }
    
    public ArrayList<Tema> listarTodos(String titulo, Integer limite){
        Tema tema = new Tema();
        tema.setTitulo(titulo);
        return this.temaDAO.listarTodos(tema, limite);
    }
    
    public Tema obtenerPorId(Integer idTema){
        return this.temaDAO.obtenerPorId(idTema);
    }
    
    
    //-------------------APARTIR DE AQUI AGREGO
    
    public ArrayList<Tema> buscarTemaPorNombre(String titulo, Integer limite){
        Tema tema = new Tema();
        tema.setTitulo(titulo);
        return this.temaDAO.listarTodos(tema,limite);
    }
    public Integer insertarTema(String titulo){
        Tema temaPorRegistrar = new Tema(null,titulo);
        Tema temaVerificacion = new Tema();
        temaVerificacion.setTitulo(titulo);
        Tema resultadoVerif = this.temaDAO.obtenerPorAtributosUnicos(temaVerificacion);
        if(resultadoVerif != null){
            return resultadoVerif.getIdTema();
        }else{
            return this.temaDAO.insertar(temaPorRegistrar);
        }        
    }
    
    public ArrayList<Tema> listarTodos_TemasDeTesisPorID(Integer idTesis){
        Tesis tesis = new Tesis();
        tesis.setIdTesis(idTesis);
        return this.tesis_temaDAO.listarTodos_TemasDeTesis(tesis,null, null);
    }
    
    public ArrayList<Tema> listarTodos_TemasDeTesis(Integer idTesis, String ts_titulo, String fechaPublicacion,
                                                    String descripcion, Integer hayFormatoFisico, Integer hayFormatoVirtual,
                                                    Integer idBibliotecaAsociada, String gradoAcademico, String tm_titulo) {
        Date d_fechaPublicacion = vkf.toValidDate(fechaPublicacion);
        Boolean b_hayFormatoFisico = vkf.toValidBoolean(hayFormatoFisico);
        Boolean b_hayFormatoVirtual = vkf.toValidBoolean(hayFormatoVirtual);
        Biblioteca bibliotecaAsociada = new Biblioteca();
        bibliotecaAsociada.setIdBiblioteca(idBibliotecaAsociada);
        TipoGrado tg_gradoAcademico = vkf.toValidEnum(gradoAcademico, TipoGrado.class);
        Tesis tesis = new Tesis(idTesis,ts_titulo, d_fechaPublicacion, descripcion,null,
                                b_hayFormatoFisico,b_hayFormatoVirtual, null, null,
                                bibliotecaAsociada,tg_gradoAcademico);
        Tema tema = new Tema(null,tm_titulo);
        return this.tesis_temaDAO.listarTodos_TemasDeTesis(tesis, tema,null);
    }
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPublicacionBO/TemaBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

namespace SoftBibliotecaPublicacionBO
{
    public class TemaBO : BaseBO
    {
        /*
       public int insertar(string titulo)
       {
           return this.WsCliente.tema_insertar(titulo);
       }

       public int modificar(int idTema, string titulo)
       {
           return this.WsCliente.tema_modificar(idTema,titulo);
       }

       public int eliminar(int idTema)
       {
           return this.WsCliente.tema_eliminar(idTema);
       }

       public BindingList<tema> listarTodos()
       {
           try
           {
               tema[] temas = this.WsCliente.tema_listarTodos();
               if (temas == null) return new BindingList<tema>();
               return new BindingList<tema>(temas);
           }
           catch (Exception ex)
           {
               return new BindingList<tema>();
           }

       }

       public BindingList<tema> listarTodos(string titulo)
       {
           try
           {
               tema[] temas = this.WsCliente.tema_listarTodos_porParametros(titulo);
               if (temas == null) return new BindingList<tema>();
               return new BindingList<tema>(temas);
           }
           catch (Exception ex)
           {
               return new BindingList<tema>();
           }
       }

       public tema obtenerPorId(int idTema)
       {
           return this.WsCliente.tema_obtenerPorId(idTema);
       }
       */


        public BindingList<tema> buscarTemaPorNombre(string titulo, int lim)
        {
            try
            {
                tema[] temas = this.WsCliente.buscarTemaPorNombre(titulo,lim);
                if (temas == null) return new BindingList<tema>();
                return new BindingList<tema>(temas);
            }
            catch (Exception ex)
            {
                return new BindingList<tema>();
            }
        }
        public int insertarTema(string titulo)
        {
            return this.WsCliente.insertarTema(titulo);
        }

        public BindingList<tema> listarTodos_TemasDeTesisPorID(int idTesis)
        {
            try
            {
                tema[] temas = this.WsCliente.listarTodos_TemasDeTesisPorID(idTesis);
                if (temas == null) return new BindingList<tema>();
                return new BindingList<tema>(temas);
            }
            catch (Exception ex)
            {
                return new BindingList<tema>();
            }
        }

        public BindingList<tema> listarTodos_TemasDeTesis(int idTesis, string ts_titulo, string fechaPublicacion,
                                                        string descripcion, int hayFormatoFisico, int hayFormatoVirtual,
                                                        int idBibliotecaAsociada, string gradoAcademico, string tm_titulo)
        {
            try
            {
                tema[] temas = this.WsCliente.listarTodos_TemasDeTesis(idTesis, ts_titulo, fechaPublicacion,
                                                        descripcion, hayFormatoFisico, hayFormatoVirtual,
                                                        idBibliotecaAsociada, gradoAcademico, tm_titulo);
                if (temas == null) return new BindingList<tema>();
                return new BindingList<tema>(temas);
            }
            catch (Exception ex)
            {
                return new BindingList<tema>();
            }
        }
    }
}
